kong/pdk/vault.lua

```diff
@@ -220,39 +295,38 @@ local function new(self)
           strategy.init()
         end
 
-        schema = schema.fields.config
+        schema = Schema.new(schema.fields.config)
       end
 
       STRATEGIES[name] = strategy
       SCHEMAS[name] = schema
     end
 
-    local config = CONFIGS[name]
-    if not config then
-      config = opts.config or {}
+    -- base config stays the same so we can cache it
+    local base_config = CONFIGS[name]
+    if not base_config then
+      base_config = {}
       if self and self.configuration then
         local configuration = self.configuration
-        local fields = schema.fields
         local env_name = gsub(name, "-", "_")
-        for i = 1, #fields do
-          local k, f = next(fields[i])
-          if config[k] == nil then
-            local n = lower(fmt("vault_%s_%s", env_name, k))
-            local v = configuration[n]
-            if v ~= nil then
-              config[k] = v
-            elseif f.required and f.default ~= nil then
-              config[k] = f.default
-            end
+        for k, f in schema:each_field() do
+          local n = lower(fmt("vault_%s_%s", env_name, gsub(k, "-", "_")))
```

Comment:
Describe `n` please. Can this even be a separate function?

Reply:
I added a comment there.